schema_based_on_diagram.sql

```diff
@@ -0,0 +1,44 @@
+/* Database schema to keep the structure of entire database. */
+
+CREATE TABLE patients (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(100) NOT NULL,
+    date_of_birth DATE NOT NULL
+);
+
+CREATE TABLE medical_histories (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP NOT NULL,
+    patient_id INT NOT NULL REFERENCES patients(id),
+	status VARCHAR(100) NOT NULL
+);
+
+CREATE TABLE treatments (
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    type VARCHAR(100) NOT NULL ,
+	name VARCHAR(100) NOT NULL
+);
+
+CREATE TABLE invoices(
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    total_amount DECIMAL NOT NULL,
+    generated_at TIMESTAMP NOT NULL,
+    payed_at TIMESTAMP NOT NULL,
+    medical_history_id INT NOT NULL REFERENCES medical_histories(id),
+);
+
+CREATE TABLE invoice_items(
+    id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    unit_price DECIMAL NOT NULL,
+    quantity INT NOT NULL,
+    total_price DECIMAL NOT NULL,
+    invoice_id INT NOT NULL,
+    treatment_id INT NOT NULL REFERENCES treatments(id),
+);
+
+
+-- join table
+CREATE TABLE treatments_history(
+    medical_history_id INT REFERENCES medical_histories(id),
+    treatment_id INT REFERENCES treatments(id)
+);
```

Comment:
- [ ] Good job with your schema, but you need to implement indexes as well, so please, add the necessary indexes to this file for each table. You can do so by typing something like `CREATE INDEX ON medical_histories (patient_id)`